### PR TITLE
Make `psql` function check the `url` key from the database YAML

### PR DESCRIPTION
**Pull request by twe4ked**
Follow up from #4!

### CODE REVIEW

File: shell/aliases/rails.sh [hunk 1]
```diff
@@ -122,6 +122,9 @@ function psql
   if [[ "$(__database_yml adapter)" == 'postgresql' ]]; then
     PGDATABASE="$(__database_yml database)" command psql "$@"
     return $?
+  elif [[ "$(__database_yml url)" =~ '^postgres' ]]; then
```
Comment (jasoncodes) [inline shell/aliases/rails.sh]: In my experience, at least on bash 3.x which comes with macOS and bash 5.x from Homebrew, the right hand side of `=~` is taken literally and should not be quoted. Is this working for you? What shell are you on?

```console
$ [[ "$(echo postgresql://foo)" =~ '^postgres' ]]; echo $?
1
$ [[ "$(echo postgresql://foo)" =~ ^postgres ]]; echo $?
0
```

File: shell/aliases/rails.sh [hunk 2]
```diff
@@ -122,6 +122,9 @@ function psql
...
+    command psql "$(__database_yml url)" "$@"
+    return $?
   fi
   command psql "$@"
 }
```
Comment (jasoncodes) [inline shell/aliases/rails.sh]: With both `url` and `adapter`+`database` defined in `database.yml`, Rails seems to prefer `url`. I feel like we should do the same.

Interestingly, it seems `url` in `database.yml` is preferred over the `DATABASE_URL` environment variable. I expected `DATABASE_URL` to always take preference.

What about something like this?

```sh
function psql
{
   if [[ "$(__database_yml url)" =~ ^postgres ]]; then
     command psql "$(__database_yml url)" "$@"
     return $?
   elif [[ "$DATABASE_URL" =~ ^postgres ]]; then
     command psql "$DATABASE_URL" "$@"
     return $?
   elif [[ "$(__database_yml adapter)" == 'postgresql' ]]; then
     PGDATABASE="$(__database_yml database)" command psql "$@"
     return $?
   fi
   command psql "$@"
 }
```

File: shell/aliases/rails.sh (hunk 1 above)
Comment (twe4ked) [reply shell/aliases/rails.sh]:
```
$ [[ "$(echo postgresql://foo)" =~ '^postgres' ]]; echo $?
0
$ [[ "$(echo postgresql://foo)" =~ ^postgres ]]; echo $?
0
$ echo "$SHELL"
/bin/zsh
```

File: shell/aliases/rails.sh (hunk 2 above)
Comment (twe4ked) [reply shell/aliases/rails.sh]: Love it!

File: shell/aliases/rails.sh (hunk 1 above)
Comment (jasoncodes) [reply shell/aliases/rails.sh]: Ah, of course: it’s another difference between bash and zsh.

**Comment by jasoncodes** [comment]
Thanks for the PR!

**Comment by twe4ked** [comment]
Thanks for getting that over the line :)

**Pull request merged**